java-snapshot-testing-core/src/test/java/au/com/origin/snapshots/DebugSnapshotTest.java

```diff
@@ -52,41 +52,41 @@ void createDebugFile(TestInfo testInfo) {
     assertTrue(Files.exists(Paths.get(DEBUG_FILE_PATH)));
   }
 
-  @DisplayName("Debug file should not be created when snapshots match for a new snapshot")
+  @DisplayName("Debug file should be created when snapshots match for a new snapshot")
   @Test
-  void noDebugFileCreatedNewSnapshot(TestInfo testInfo) {
+  void debugFileCreatedNewSnapshot(TestInfo testInfo) {
     assertTrue(Files.exists(Paths.get(SNAPSHOT_FILE_PATH)));
 
     SnapshotVerifier snapshotVerifier = new SnapshotVerifier(DEFAULT_CONFIG, testInfo.getTestClass().get());
     Expect expect = Expect.of(snapshotVerifier, testInfo.getTestMethod().get());
     assertTrue(Files.notExists(Paths.get(DEBUG_FILE_PATH)));
     expect.toMatchSnapshot(new TestObjectGood());
-    assertTrue(Files.notExists(Paths.get(DEBUG_FILE_PATH)));
+    assertTrue(Files.exists(Paths.get(DEBUG_FILE_PATH)));
   }
 
-  @DisplayName("Debug file should not be created when snapshots match for an existing snapshot")
+  @DisplayName("Debug file should be created when snapshots match for an existing snapshot")
   @Test
-  void noDebugFileCreatedExistingSnapshot(TestInfo testInfo) {
+  void debugFileCreatedExistingSnapshot(TestInfo testInfo) {
     assertTrue(Files.exists(Paths.get(SNAPSHOT_FILE_PATH)));
 
     SnapshotVerifier snapshotVerifier = new SnapshotVerifier(DEFAULT_CONFIG, testInfo.getTestClass().get());
     Expect expect = Expect.of(snapshotVerifier, testInfo.getTestMethod().get());
     assertTrue(Files.notExists(Paths.get(DEBUG_FILE_PATH)));
     expect.toMatchSnapshot(new TestObjectGood());
-    assertTrue(Files.notExists(Paths.get(DEBUG_FILE_PATH)));
+    assertTrue(Files.exists(Paths.get(DEBUG_FILE_PATH)));
```

Comment:
Agree, but I will have to do it after validate snapshots as mentioned above.

Reply:
Hi @mmiesiak 

Are you updating to delete the debug snapshots if they match exactly with the original?

Reply:
Hi @jackmatt2 

I had less time to work on that during last time. 

I have added what you asked for but it's not covered by unit tests yet. 

I also fixed spock snapshot extension so it calls `SnapshotVerifier.validateSnapshots()` in CLEANUP_SPEC phase.

Reply:
Hi @jackmatt2 

The tests exist in `EqualDebugSnapshotFileTest.java`.

I had to change the implementation to compare files content because it was easier to write the tests this way. I could also revert to previous implementation and create a test that calls twice `Expect.toMatchSnapshot` with different methods and later call `SnapshotVerifier.validateSnapshots()` but I find the current implemention cleaner.

Thanks for your patience.

Reply:
@mmiesiak @jackmatt2 It seems there is some issue with 3.3.0 version leaving debug files around.
Reported it here: https://github.com/origin-energy/java-snapshot-testing/issues/99

Reply:
right ... I thought this was resolved as part of this PR.